pkg/registry/configurator.go

```diff
@@ -28,13 +35,29 @@ type RegistryConfig struct {
 	ExternalRegistry *Registry
 }
 
-func NewConfigurator(conf RegistryConfig, containerRuntimeInfo containerruntime.Info) (Configurator, error) {
+func NewConfigurator(conf RegistryConfig, containerRuntimeInfo containerruntime.Info, infraDriver infradriver.InfraDriver) (Configurator, error) {
+	rootfs := infraDriver.GetClusterRootfs()
+
+	var containerRuntimeConfigurator containerruntime.Configurator
+
+	if containerRuntimeInfo.Type == "docker" {
+		containerRuntimeConfigurator = containerruntime.NewDockerRuntimeDriver(infraDriver)
+	}
+
+	if containerRuntimeInfo.Type == "containerd" {
+		containerRuntimeConfigurator = containerruntime.NewContainerdRuntimeDriver(infraDriver)
+	}
+
 	if conf.LocalRegistry != nil {
 		return &localSingletonConfigurator{
-			LocalRegistry:        *conf.LocalRegistry,
-			ContainerRuntimeInfo: containerRuntimeInfo,
+			rootfs:                       rootfs,
+			LocalRegistry:                conf.LocalRegistry,
+			configFileGenerator:          NewLocalFileGenerator(rootfs),
```

Comment:
I think if one filed can be init inside of localSingletonConfigurator, maybe we should not configure it outside.

configFileGenerator and rootfs.